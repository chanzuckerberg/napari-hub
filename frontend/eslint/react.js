module.exports = {
  extends: ['airbnb/hooks'],

  plugins: ['napari-hub'],

  rules: {
    'napari-hub/no-root-mui-import': 'error',

    /*
      Prop types aren't necessary since we have TypeScript interfaces for
      prop types.
    */
    'react/prop-types': 'off',

    /*
      React automatically adds the import as part of the new JSX transform:
      https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#nextjs
     */
    'react/react-in-jsx-scope': 'off',

    /*
      Prop spreading is very useful as a simple way for passing multiple
      props to a component. However, we should still be careful about passing
      unnecessary props.
    */
    'react/jsx-props-no-spreading': 'off',

    /*
      This rule isn't really needed anymore since we can use ES6 object defaults
      for props. There are also plans to deprecate the use of `defaultProps`:
      https://github.com/reactjs/rfcs/pull/107
    */
    'react/require-default-props': 'off',
  },
};
