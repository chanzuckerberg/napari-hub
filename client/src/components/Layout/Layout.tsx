import { ReactNode } from 'react';

interface Props {
  children: ReactNode;
}

export function Layout({ children }: Props) {
  return <main>{children}</main>;
}
