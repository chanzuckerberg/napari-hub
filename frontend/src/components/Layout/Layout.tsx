import { ReactNode } from 'react';

import { AppBar } from '@/components/AppBar';
import { Footer } from '@/components/Footer';
import { SignupForm } from '@/components/SignupForm';

interface Props {
  children: ReactNode;
}

export function Layout({ children }: Props) {
  return (
    <div className="flex flex-col min-h-screen">
      {!process.env.PREVIEW && <AppBar />}

      <main className="flex flex-col flex-grow">{children}</main>
      <SignupForm />
      <Footer />
    </div>
  );
}
