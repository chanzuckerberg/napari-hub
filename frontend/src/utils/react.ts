import { ReactElement, ReactNode } from 'react';

/**
 * Checks if a React node is an element. An element is an instantiated
 * component.
 *
 * @param node The React node
 * @returns True if node is an element
 */
export function isReactElement(node: ReactNode): node is ReactElement {
  const element = node as ReactElement | null | undefined;
  return !!(element?.type && element?.props);
}
