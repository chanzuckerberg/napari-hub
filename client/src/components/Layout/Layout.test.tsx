import { render, RenderResult, screen } from '@testing-library/react';
import { ReactNode } from 'react';

import { Layout } from './Layout';

describe('<Layout />', () => {
  const text = 'Hello, World!';
  let component: RenderResult;
  let child: ReactNode;

  beforeEach(() => {
    child = <h1>{text}</h1>;
    component = render(<Layout>{child}</Layout>);
  });

  it('should match snapshot', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render child', () => {
    const renderedChild = screen.getByText(text);
    expect(renderedChild).toBeTruthy();
  });
});
