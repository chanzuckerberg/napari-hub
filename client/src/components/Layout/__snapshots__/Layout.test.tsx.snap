// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed top-0 right-0 flex flex-row items-start z-40 w-9/12 h-screen p-6 bg-black"
    data-testid="menu"
    style="transform: translateX(100%) translateZ(0);"
  >
    <ul
      class="flex flex-auto flex-col"
    >
      <li
        class="text-white item"
        data-testid="drawerItem"
      >
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="text-white item"
        data-testid="drawerItem"
      >
        <a
          href="/help"
        >
          Help
        </a>
      </li>
    </ul>
    <button
      class="flex"
      data-testid="drawerClose"
      type="button"
    >
      <svg
        fill="none"
        height="18"
        viewBox="0 0 18 18"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1L17 17"
          stroke="white"
          stroke-width="2.35294"
        />
        <path
          d="M17 1L1 17"
          stroke="white"
          stroke-width="2.35294"
        />
      </svg>
    </button>
  </div>
  <nav
    class="bg-napari-primary h-napari-app-bar px-6 md:px-12 2xl:p-0 grid grid-cols-napari-nav-mobile justify-center items-center gap-6 md:gap-12 2xl:grid 2xl:grid-cols-napari-app-bar-2-col 3xl:grid 3xl:grid-cols-napari-3-col"
  >
    <header
      class="flex"
      data-testid="appBarHeader"
    >
      <h1
        class="whitespace-nowrap"
      >
        <a
          href="/"
        >
          napari 
          <strong>
            hub
          </strong>
        </a>
      </h1>
    </header>
    <div
      class="flex items-center w-full max-w-napari-center-col justify-self-end"
    >
      <form
        class="flex flex-auto items-center border-b-2 border-black"
      >
        <input
          class="flex flex-auto border-none outline-none bg-transparent w-0"
          data-testid="searchBarInput"
          value=""
        />
        <div
          style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 14px; height: 14px;"
        >
          <noscript />
          <img
            alt="Icon for napari search bar"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </form>
      <ul
        class="hidden lg:flex ml-12 links"
      >
        <li
          class="list-none"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="list-none"
        >
          <a
            href="/help"
          >
            Help
          </a>
        </li>
      </ul>
      <button
        class="ml-6 flex lg:hidden"
        type="button"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Menu button icon"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </button>
    </div>
  </nav>
  <main>
    <h1>
      Hello, World!
    </h1>
  </main>
</DocumentFragment>
`;
